joomla/joomla-cms
# [5.1] TUF Code Cleanup

Various minor reworkings to the TUF code as below

### Summary of Changes
- Removes the extra HTTP Factory - it serves no purpose. Whilst we don't implement PSR-18 yet it shows the type of object to be passed around which is a created HTTP client - not the factory for it: https://www.php-fig.org/psr/psr-18/ 
- Rework constructor args - db is now required (it already existed in one of the two instantiating classes anyhow and we call Factory::getDbo() already in the other), makes application a required parameter (we were statically grabbing it in the class - which shouldn't be the case for new classes).
- Fixes the constructor doc block

### Testing Instructions
Check all TUF functionality from https://github.com/joomla/joomla-cms/pull/42799 continues to work with no changes. New code is slightly easier to maintain and read.

### Link to documentations
Please select:
